iwyu_ast_util.cc

```diff
@@ -1022,11 +1022,9 @@ template <class T> inline set<const clang::NamedDecl*> GetRedeclsOfRedeclarable(
 // The only way to find out whether a decl can be dyn_cast to a
 // Redeclarable<T> and what T is is to enumerate the possibilities.
 // Hence we hard-code the list.
-set<const clang::NamedDecl*> GetNonclassRedecls(const clang::NamedDecl* decl) {
-  CHECK_(!isa<RecordDecl>(decl) && "For classes, call GetClassRedecls()");
-  CHECK_(!isa<ClassTemplateDecl>(decl) && "For tpls, call GetClassRedecls()");
-  if (const TagDecl* specific_decl = DynCastFrom(decl))
```

Comment:
[The only place](https://github.com/include-what-you-use/include-what-you-use/pull/1059/files/fc44f68622c1ff386addf3beaa89f4843745d739#diff-96c9906a158b909fda0f001cb5bd04615d46704df24173306894f3d228abbed1R1202) where it is used is resolving whether a full symbol use should not be reported due to presence of the definition in the same file with the use. For classes and templates, only one definition should be present. For functions and global variables, any redeclaration could be considered as a definition "in the IWYU sense". Previously, enumerations fell into the second category (and it was right because enumerators were reported separately). But now, they fall into the first category, because they are treated almost similarly to classes.

Reply:
Nice, thanks for the extra explanation.